packages/react-components/src/theme-provider/src/ThemeContext.ts

```diff
@@ -1,13 +1,13 @@
 import { createContext, useContext } from "react";
-import type { ColorScheme, Theme } from "./ThemeProvider";
+import type { ColorScheme, ColorSchemeOrSystem, Theme } from "./ThemeProvider";
 
 export interface ThemeContextType {
-    theme?: Theme;
-    colorScheme?: ColorScheme;
-    setColorScheme?: (newColorScheme: ColorScheme) => void;
+    theme: Theme;
+    colorScheme: ColorScheme;
+    setColorScheme: (newColorScheme: ColorSchemeOrSystem) => void;
 }
 
-export const ThemeContext = createContext<ThemeContextType>({});
+export const ThemeContext = createContext<ThemeContextType>({} as ThemeContextType);
```

Comment:
I don't get why we need to cast here since we don't have to for other similar cases like 

```
export const DialogTriggerContext = createContext<DialogTriggerContextType>({});

export function useDialogTriggerContext() {
    return useContext(DialogTriggerContext);
}
```

Reply:
It means that all the property in DialogTriggerContextType are optional. 

I beleive that the object passed here is used in the case where someone would call useThemeContext without being in a ThemeContext.Provider. it would fallback on those value. 

Since we don't support this case, i forced the type in this situation

Reply:
So the context values should remain optional then? If the hook can be called without having a parent ThemeContext.

Reply:
Basically the empty object literal is to ensure it could be destructured without having to check against an undefined value.

Reply:
Chakra does the same for their color mode provider https://github.com/chakra-ui/chakra-ui/blob/326813ea261ddb6c9f0106e7e63f49f51e111fa2/packages/color-mode/src/color-mode-provider.tsx#L25

Reply:
It's a pattern that have been generalized and was working fine without TS.

With TS, I believe those default empty object literals should be replaced by undefined (well no default value) and it should be the responsability of the consumer to ensure it's not undefined before destructuring?

Therefore the context props could be required.

That being said, I would prefer to do it accross the whole codebase not only for this usecase. I can create an issue for this.

I doubt this typing really block your migration?

Reply:
This is not blocking, so i don't mind looking at a solution that would be good for the whole codebase